### PR TITLE
Globally calculated parameter in edit views

### PR DESCRIPTION
**Pull request by Elfangor93**
This PR adds the global calculated parameter in the Parameters tab of the "Edit Image" and "Edit Category" view. The dynamic calculation of the global parameter is done with the same logic which will later be applied but without taking into account parameters set by the Menu Item. In addition, the parameters are carried out on the basis of the current user. Therefore the calculated value can change based on the user group you are in...

grafik

### CODE REVIEW

**Comment by eumel1602** [comment]
If I click on the two links: Configuration inheritance and Configuration Set in an image or category, the respective setting options are shown in the same window. If I then change something there and save it, I come back to the main overview (images or category). The edited image/category is then locked with a lock. This could no longer allow users without superadmin rights to edit the image?
Wouldn't it be better if the two links open a modal window and after changing the settings you are back at the selected image/category?

(Otherwise the PR shouldn't do anything except the link, right?)
sorry for always asking questions like this, but unfortunately it doesn't explain what to do here...
zzzbild

**Comment by Elfangor93** [comment]
No, the new functionality is the value in brackets
grafik

Based on the chosen Configuration inheritance method and your settings in the configuration sets, this value in brackets gets calculated and is then shown.

**Comment by Elfangor93** [comment]
>If I click on the two links: Configuration inheritance and Configuration Set in an image or category, the respective setting options are shown in the same window. If I then change something there and save it, I come back to the main overview (images or category). The edited image/category is then locked with a lock.

I can not confirm that. Can you please provide step by step how to reproduce this issue?

**Comment by MrMusic** [comment]
>If I click on the two links: Configuration inheritance and Configuration Set in an image or category, the respective setting options are shown in the same window. If I then change something there and save it, I come back to the main overview (images or category). The edited image/category is then locked with a lock.

>I can not confirm that. Can you please provide step by step how to reproduce this issue?

If you are in the parameters tab and click on one of the two links in the description , the currently edited category remains as "Checked out".
Recommendation: remove the link.

parameters-tab


I have only tested categories:

For me, the displayed value does not correspond to the value from the configuration but to the last saved setting of the respective category.
Example: The setting for Allow Rating in the configuration is 'Yes'.
Open a category, display for Allow Rating is "Use Global (Yes)" ->Correct.
Change the setting to 'No' and save the category.
Now Display for Allow Rating is "Use Global (No) ->Wrong

In addition, some of the parameters no longer make sense, as some things have changed in comparison to JoomGallery 3.

Allow Rating - Yes, you can adopt this 1:1.

Add Watermark
 -This refers to the 'dynamic' watermark or 'fixed' watermark or both?
 -There is now a separate option to set the watermark for each image type. Which image type should be used for "Use Global"?
   Should the value set in the category override the setting for all image types?

Offer Downloads - Yes, you can adopt this 1:1.

Download with Watermark
 -Same problem as with Add watermark

The display would have to be adapted somewhat, currently it is difficult to see which value is selected, see screenshot.

**Comment by Elfangor93** [comment]
>In addition, some of the parameters no longer make sense, as some things have changed in comparison to JoomGallery 3.

Yes, you are right. Maybe we should discuss this in our next online meeting.

We have now the possibility to override parameters on ...
- ... categories (they will apply for all images and sub-categories of this category)
- ... images (they will apply specifically for this image)
- ... menu item (they will apply for all images and categories displayed within the view of this menu item)

**Comment by reilldesign** [comment]
A little late, but still important. To ensure a consistent user experience a standard display with the Joomla dropdown solution would be desirable (screenshot).

JDropdown

**Comment by Elfangor93** [comment]
>A little late, but still important. To ensure a consistent user experience a standard display with the Joomla dropdown solution would be desirable (screenshot).

If the other agree with @reilldesign to use a dropdown field instead of the currently used buttons then please open a separate issue for that. This change will produce a rat taile of changes and can be improved in a later time.

**Comment by AlexanderSupp** [comment]
There are some confusing blemishes. I will put these together when we have achieved a first degree of completion of the most important functions.
I agree with @reilldesign.

**Comment by Elfangor93** [comment]
@reilldesign Since this is an enhancement and high priority I added an issue for your request to discuss and implement that later on.

**Comment by MrMusic** [comment]
I think the shown global calculated parameter is still wrong.
@Elfangor93  Did you take a look at my code reviews?

**Comment by MrMusic** [comment]
>If you dont finish your review, the comments are only visible to you...

Thanks! 
I have misunderstood this.

File: administrator/com_joomgallery/layouts/joomla/form/field/radio/configbtns.php [hunk 1]
```diff
@@ -133,22 +133,22 @@
...
                     <?php
-                      $value = false;
+                      $val = false;
                       if($config)
                       {
-                        $value = $comp->getConfig()->get($configName, false);
+                        $val = $comp->getConfig()->get($configName, false);
                       }
-                      $value = $value ? Text::_('JYES') : Text::_('JNO');
+                      $val = $value ? Text::_('JYES') : Text::_('JNO');
```
[the remark this answers is not in the document]
Comment (Elfangor93) [reply administrator/com_joomgallery/layouts/joomla/form/field/radio/configbtns.php:149]: What do you mean??
Changing `$value` to `$val` fixes the display of the currently selected option. Its the fix for the display issue you reported here: https://github.com/JoomGalleryfriends/JG4-dev/pull/106#issuecomment-1587335679

[the remark this answers is not in the document]
Comment (MrMusic) [reply administrator/com_joomgallery/layouts/joomla/form/field/radio/configbtns.php:149]: The error is:
_For me, the displayed value does not correspond to the value from the configuration but to the last saved setting of the respective category.
Example: The setting for Allow Rating in the configuration is 'Yes'.
Open a category, display for Allow Rating is "Use Global (Yes)" ->Correct.
Change the setting to 'No' and save the category.
Now Display for Allow Rating is "Use Global (No) ->Wrong_

I suspect the line 149 should correctly look like this:
`$val = $val ? Text::_('JYES') : Text::_('JNO');`

[the remark this answers is not in the document]
Comment (Elfangor93) [reply administrator/com_joomgallery/layouts/joomla/form/field/radio/configbtns.php:149]: Yes, but it does look like this...

[the remark this answers is not in the document]
Comment (Elfangor93) [reply administrator/com_joomgallery/layouts/joomla/form/field/radio/configbtns.php]: Oke, got it. There is a type ;-)

**Comment by MrMusic** [comment]
Parameters are good now. Thanks.
Design is discussed [here](https://github.com/JoomGalleryfriends/JG4-dev/issues/112).
I have tested this item ✅ successfully. Thank you very much.

**Pull request merged**